src/Type/CallableType.php

```diff
@@ -188,6 +191,32 @@ public function getReturnType(): Type
 		return $this->returnType;
 	}
 
+	public function inferTemplateTypes(Type $receivedType): TemplateTypeMap
+	{
+		if ($receivedType instanceof UnionType || $receivedType instanceof IntersectionType) {
+			return $receivedType->inferTemplateTypesOn($this);
+		}
+
+		if ($receivedType->isCallable()->no()) {
+			$parametersAcceptor = new TrivialParametersAcceptor();
+		} else {
+			$parametersAcceptor = ParametersAcceptorSelector::selectSingle($receivedType->getCallableParametersAcceptors(new OutOfClassScope()));
```

Comment:
I think it's safer to go over all parameters acceptors. This way, it can break when we construct a callable to a function/methods that has multiple variants.